Beginner Vicmap tutorial

rachelannieswain opened the pull request:
Adds tutorial for R users with minimal spatial experience to perform basic plotting and spatial manipulation with Vicmap data.

Includes short introduction of the Vicmap catalogue and uses the swooping birds dataset to provide worked examples.

Commit:
Transfer tutorial to vicmapr vignette

Commit:
First review of vicmap tutorial

Commit:
Reviewed tutorial and builds site

JustinCally commented:
Can you add a section that describes how to use the CQL Geometry Predicates (e.g. https://justincally.github.io/VicmapR/articles/query_vicmap.html#geometric-filters)? This might be preferable to geometric filtering with sf after the results are collected

Commit:
Tests use of geometric filters

Commit:
Adds geometric filter examples

Commit:
Minor review, updates complete

rachelannieswain commented:
Added a section that repeats exercises using sf to only use VicmapR geometric filters. 

Currently having issues building the pkgdown site. I suspect this is due to library version issues on my end causing the VicmapR queries with geometric filters to not work consistently. Would be interested to know if you can build the site at your end.

Commit:
Pulls master and rebuilds pkgdown site

Commit:
Adds VignetteBuilder field and documents route dataset

Commit:
Rebuilds site

Commit:
Rebuilds site

Commit:
Suggests kableExtra

Commit:
Rebuilds site

Commit:
Suggests mapview

Commit:
Trigger new checks after Vicmap server down

Commit:
Changes route data ref

Commit:
References example data as geojson in /shapes

Commit:
Check gdal version on github

Commit:
Rebuilds site

Commit:
Display gdal version

Commit:
Try Warning

Commit:
Use polygon for spatial filter

Commit:
Updates setup chunk and adds licence details

Commit:
Updates description with vignette pkgs

Commit:
Rebuilds site

Commit:
Checks gdal and cran status

Commit:
Rebuilds site

Commit:
Install latest sf on OS

adam-gruer commented on .github/workflows/R-CMD-check.yaml:
```diff
@@ -64,6 +64,7 @@ jobs:
         if: runner.os == 'macOS'
         continue-on-error: true
         run: |
+          Rscript -e "install.packages('sf')"
```
@rachelannieswain  I think this line should be placed after the brew install lines as sf will need those libraries  before it can be installed

Commit:
Reorder sf install

Commit:
Update sf ubuntu

Commit:
Installs mapview

JustinCally committed:
order of gdal update

JustinCally committed:
remove ubuntugis update

JustinCally committed:
try rocker geospatial

JustinCally committed:
update cache

JustinCally committed:
remove docker

JustinCally committed:
update cache

JustinCally committed:
only evaluate chunks if gdal > 3

JustinCally committed:
commas

JustinCally committed:
r versiot and vignette builder

JustinCally committed:
try with knitr notangle

JustinCally committed:
knitr output

JustinCally committed:
purl = FALSE

JustinCally committed:
wrong purl location

JustinCally committed:
fix vignette

JustinCally committed:
typo

JustinCally committed:
echo = false

JustinCally committed:
setip chunk evals first

JustinCally committed:
rmarkdown output

JustinCally committed:
other vignette changed

JustinCally committed:
remove inline eval

JustinCally committed:
remove notangle

JustinCally committed:
dbplyr compat

Merge branch 'master' into beginner_vicmap_tutorial
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

JustinCally reviewed (approved):
Looks good. Check: https://justincally.github.io/VicmapR/dev/ Once merged and uploaded. Can edit minor issues if there are any problems.
